### PR TITLE
Update `_replace_obs`, `_make_probs`, and `algebra_commutator` batch transforms

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Update the batch transforms to the new transform API.

**Description of the Change:**
Update `_replace_obs`, `_make_probs`, and `algebra_commutator`.

### CODE REVIEW

File: pennylane/optimize/riemannian_gradient.py [hunk 1]
```diff
@@ -69,7 +69,6 @@ def append_time_evolution(tape, riemannian_gradient, t, n, exact=False):
         qml.apply(obj)
 
 
-@qml.batch_transform
 def algebra_commutator(tape, observables, lie_algebra_basis_names, nqubits):
```
Comment (eddddddy) [inline pennylane/optimize/riemannian_gradient.py:72]: This is just a helper function for the `RiemannianGradientOptimizer` and doesn't need to be a batch transform.

Comment (rmoyard) [reply pennylane/optimize/riemannian_gradient.py:72]: Great one less :+1:

File: pennylane/interfaces/execution.py [hunk 2]
```diff
@@ -419,6 +419,7 @@ def cost_fn(params, x):
             device,
             gradient_fn,
             interface=interface,
+            transform_program=transform_program,
```
Comment (rmoyard) [inline pennylane/interfaces/execution.py:422]: Good catch :+1:

File: tests/shadow/test_shadow_transforms.py [hunk 3]
```diff
@@ -98,6 +98,29 @@ def circuit(x, obs):
     return circuit
 
 
+class TestReplaceObs:
```
Comment (rmoyard) [inline tests/shadow/test_shadow_transforms.py:101]: Thanks for adding it :+1:

File: tests/shadow/test_shadow_transforms.py [hunk 4]
```diff
@@ -308,22 +331,6 @@ def test_backward_torch(self):
 class TestExpvalTransform:
     """Test that the expval transform is applied correctly"""
 
-    def test_hadamard_transform(self):
```
Comment (rmoyard) [inline tests/shadow/test_shadow_transforms.py:311]: Why was this test removed?

**Review by rmoyard — COMMENTED** [review]
All good to me, just wondering why you remove the shadow expval test?

File: tests/shadow/test_shadow_transforms.py (hunk 4 above)
Comment (eddddddy) [reply tests/shadow/test_shadow_transforms.py:311]: This test only checks that the `_replace_obs` transform works correctly. I added these tests above.

**Review by rmoyard — APPROVED** [review]
:100: Thanks it looks great!

**Pull request merged**